src/timedata.cpp

```diff
@@ -40,16 +40,20 @@ static int64_t abs64(int64_t n)
     return (n >= 0 ? n : -n);
 }
 
+#define BITCOIN_TIMEDATA_MAX_SAMPLES 200
+
 void AddTimeData(const CNetAddr& ip, int64_t nOffsetSample)
 {
     LOCK(cs_nTimeOffset);
     // Ignore duplicates
     static set<CNetAddr> setKnown;
+    if (setKnown.size() == BITCOIN_TIMEDATA_MAX_SAMPLES)
+        return;
```

Comment:
Wow, that's subtle. I thought the same as @jonasschnelli until re-reading the comments a few times.

Concept ACK, but I'd prefer to see it implemented in a way such that the early return depends on the quirk, otherwise I'm afraid it could introduce new bugs down the road. At the very least it deserves its own comment.